### ISSUE TITLE
OpenMM energy calculation units

### ISSUE THREAD

**Issue by jthorton** [MolSSI/QCEngine#224]
**Describe the bug**

When calculating the energy of a molecule using the OpenMM harness I get overly large energies. I think this is due to the unit conversion used resulting in hartrees/mol.

**To Reproduce**

This is an example that calculates the energy of a molecule in qcarchive using QCEngine
```python
import qcportal as ptl
from simtk import openmm, unit
from openforcefield.typing.engines.smirnoff import ForceField
from openforcefield.topology import Molecule
import qcelemental as qcel
import qcengine

# qcelemental converter
ureg = qcel.PhysicalConstantsContext("CODATA2014")

client = ptl.FractalClient()
# load a molecule from the qcarchive 
opts = client.get_collection('OptimizationDataset', 'SMIRNOFF Coverage Set 1')

# get the first entry from the set
entry = opts.get_entry(opts.df.index[0])

# grab the intial molecule
molecule = client.query_molecules(id=entry.initial_molecule)[0]
# use QCengine to compute the energy of the molecule

#set up the task
openmm_task = {"schema_name": "qcschema_input",
            "schema_version": 2,
            "molecule": molecule,
            "driver": "energy",
            "model": {"method": 'openmm', "basis": 'openff-1.0.0', 'offxml': 'openff-1.0.0.offxml'},
            "keywords": {"scf_type": "df"}}
# compue the energy
result = qcengine.compute(openmm_task, 'openmm')
print(result.return_result)
```
> -3.4178659499631775e+22
```python
# calculate again using openmm

# first make the offmol follwoing the method in qcengine
rdmol = qcengine.programs.rdkit.RDKitHarness._process_molecule_rdkit(molecule)
offmol = Molecule.from_rdkit(rdmol)
# load the force field 
ff = ForceField('openff-1.0.0.offxml')
#make the topology
off_top = offmol.to_topology()
# make the system
system = ff.create_openmm_system(off_top)
# make an integrator
integrator = openmm.VerletIntegrator(1.0 * unit.femtoseconds)
# set the platform
platform = openmm.Platform.getPlatformByName("CPU")
# create the context
context = openmm.Context(system, integrator, platform)

# Set positions from our Open Force Field `Molecule`
context.setPositions(offmol.conformers[0])

# Compute the energy of the configuration
state = context.getState(getEnergy=True)
# Get the potential as a simtk.unit.Quantity, put into units of hartree
q = state.getPotentialEnergy() / unit.hartree
print(q)
```
> -3.4178659499631775e+22 /mol

```python
# now get the energy again and convert using qcelemental
q = state.getPotentialEnergy() * ureg.conversion_factor("kJ / mol", "hartree")
print(q._value)
```
> -0.05602423977045684

**Expected behavior**

The energy should be returned in Hartrees, not hartree/mol.

**Additional context**

This will also affect the gradient conversion factor used.

**Comment by Lnaden** [comment]
In the OpenMM unit system, the `hartree` unit is defined only in terms of an "energy" dimension so its base is relative to the `joule` unit. The `mole` unit is in a separate dimension for "amount", so I don't think the simtk.unit module is intended to have the hartree interface with the "amount" dimension. 

Since the base units of `state.getPotentialEnergy()` are kJ/mol, I think you will have to to do 

```
state.getPotentialEnergy() / unit.AVOGADRO_CONSTANT_NA / unit.hartree
```

To get the correct units and conversions.

**Comment by dgasmith** [comment]
Closed by #225 and #226.
